pkg/cli/internal/cmd/e2e-test/setup-e2e-operator.go

```diff
@@ -0,0 +1,175 @@
+	"github.com/spf13/cobra"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	apiextensionsv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/apimachinery/pkg/util/wait"
+	clientgoscheme "k8s.io/client-go/kubernetes/scheme"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/config"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+
+	operatorv1alpha1 "github.com/kubermatic/kubecarrier/pkg/apis/operator/v1alpha1"
+
+	"github.com/kubermatic/kubecarrier/pkg/internal/util"
+
+	"github.com/kubermatic/kubecarrier/pkg/internal/reconcile"
+)
+
+var (
+	scheme = runtime.NewScheme()
+)
+
+func init() {
+	utilruntime.Must(clientgoscheme.AddToScheme(scheme))
+	utilruntime.Must(apiextensionsv1.AddToScheme(scheme))
+	utilruntime.Must(operatorv1alpha1.AddToScheme(scheme))
+}
+
+func newSetupE2EOperator(log logr.Logger) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "setup-e2e-operator",
+		Short: "install e2e operator in the given cluster",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			kubeconfig := cmd.Flag("kubeconfig").Value.String()
+			namespaceName := cmd.Flag("namespace").Value.String()
+			if err := setupE2EOperator(log, kubeconfig, namespaceName, cmd.OutOrStdout()); err != nil {
+				return fmt.Errorf("setup-e2e: %w", err)
+			}
+			return nil
+		},
+	}
+	cmd.Flags().String("kubeconfig", os.Getenv("KUBECONFIG"), "cluster kubeconfig where to install")
+	cmd.Flags().StringP("namespace", "n", "kubecarrier-e2e-operator", "namespace where to deploy operator")
+	return cmd
+}
+
+func setupE2EOperator(log logr.Logger, kubeconfig string, namespaceName string, output io.Writer) error {
+	namespace := &corev1.Namespace{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: namespaceName,
+		},
+	}
+	if kubeconfig != "" {
+		if err := os.Setenv("KUBECONFIG", kubeconfig); err != nil {
+			return fmt.Errorf("kubeconfig env setup: %w", err)
+		}
+	}
+	ctx := context.Background()
+	cfg, err := config.GetConfig()
+	if err != nil {
+		return fmt.Errorf("getting Kubernetes cluster config: %w", err)
+	}
+
+	c, err := client.New(cfg, client.Options{Scheme: scheme})
+	if err != nil {
+		return fmt.Errorf("creating Kubernetes client: %w", err)
+	}
+	s := wow.New(output, spin.Get(spin.Dots), "spinner text")
+
+	startTime := time.Now()
+	if err := spinner.AttachSpinnerTo(s, startTime, "creating namespace", func() error {
+		_, err := controllerutil.CreateOrUpdate(ctx, c, namespace, func() error {
+			return nil
+		})
+		return err
+
+	}); err != nil {
+		return err
+	}
+
+	if err := spinner.AttachSpinnerTo(s, startTime, "deploy e2e-operator", func() error {
+		objects, err := fakeoperator.Manifests(
+			fakeoperator.Config{
+				Namespace: namespace.Name,
```

Comment:
when is the config field name used except in the fake operator test?

Reply:
yes, only in test, should I get rid of it?

Reply:
If you're going to use is then no. If you're using it only in the tests and never in the code, then yes, delete it!